arches/app/src/components/ControlledListManager/ListTreeControls.vue

```diff
@@ -0,0 +1,389 @@
+
+import { displayedRowKey, selectedLanguageKey } from "@/components/ControlledListManager/const.ts";
+import { listAsNode } from "@/components/ControlledListManager/utils.ts";
+
+import Button from "primevue/button";
+import ConfirmDialog from "primevue/confirmdialog";
+import Dropdown from "primevue/dropdown";
+import SplitButton from "primevue/splitbutton";
+import { useConfirm } from "primevue/useconfirm";
+import { useToast } from "primevue/usetoast";
+
+import type { Ref } from "vue";
+import type { TreeExpandedKeys, TreeSelectionKeys } from "primevue/tree/Tree";
+import type { TreeNode } from "primevue/treenode";
+import type { Language } from "@/types/arches";
+import type { ControlledList, DisplayedRowRefAndSetter } from "@/types/ControlledListManager";
+
+// not user-facing
+const DANGER = "danger";
+const ERROR = "error";
+
+const { setDisplayedRow } = inject(displayedRowKey) as DisplayedRowRefAndSetter;
+const selectedLanguage = inject(selectedLanguageKey) as Ref<Language>;
+
+const controlledListItemsTree = defineModel<TreeNode[]>({ required: true });
+const expandedKeys = defineModel<TreeExpandedKeys>("expandedKeys", { required: true });
+const selectedKeys = defineModel<TreeSelectionKeys>("selectedKeys", { required: true });
+const movingItem = defineModel<TreeNode>("movingItem", { required: true });
+const isMultiSelecting = defineModel<boolean>("isMultiSelecting", { required: true });
+
+const { $gettext, $ngettext } = useGettext();
+const buttonGreen = "#10b981";
+
+const confirm = useConfirm();
+const toast = useToast();
+
+const deleteDropdownOptions = [
+    {
+        label: $gettext("Delete Multiple"),
+        command: () => {
+            isMultiSelecting.value = true;
+        },
+    },
+];
+
+const expandAll = () => {
+    for (const node of controlledListItemsTree.value) {
+        expandNode(node);
+    }
+};
+
+const collapseAll = () => {
+    expandedKeys.value = {};
+};
+
+const expandNode = (node: TreeNode) => {
+    if (node.children && node.children.length) {
+        expandedKeys.value[node.key as string] = true;
+
+        for (const child of node.children) {
+            expandNode(child);
+        }
+    }
+};
+
+const fetchLists = async () => {
+    let errorText;
+    try {
+        const response = await fetch(arches.urls.controlled_lists);
+        if (!response.ok) {
+            errorText = response.statusText;
+            const body = await response.json();
+            errorText = body.message;
+            throw new Error();
+        } else {
+            await response.json().then((data) => {
+                controlledListItemsTree.value = (data.controlled_lists as ControlledList[]).map(
+                    l => listAsNode(l, selectedLanguage.value)
+                );
+            });
+        }
+    } catch {
+        toast.add({
+            severity: ERROR,
+            summary: errorText || $gettext("Unable to fetch lists"),
+        });
+    }
+};
+
+const createList = async () => {
```

Comment:
It seems like a lot of these could be simplified and moved to `api.ts`, but I'm not sure we've landed on a decided pattern yet. Marking this as a discussion point 👍

Reply:
Yeah I was hoping to get some time to A/B a couple alternatives, including tinkering with these fetch calls in some way versus doing it using axios, but interacting with gettext might get tricky.